seleniumJSEnv/src/main/scala/org/scalajs/jsenv/selenium/AbstractSeleniumJSRunner.scala

```diff
@@ -23,15 +24,30 @@ private[selenium] abstract class AbstractSeleniumJSRunner(
   protected def console: JSConsole = _console
   protected def driver: WebDriver with JavascriptExecutor = _driver
 
-  protected def startInternal(logger: Logger, console: JSConsole): Unit = synchronized {
+  protected def setupRun(logger: Logger, console: JSConsole): Unit = synchronized {
     require(_driver == null, "start() may only start one instance at a time.")
     require(_logger == null && _console == null)
     _logger = logger
     _console = console
     _driver = factory()
   }
 
-  def stop(): Unit = synchronized {
+  protected def endRun(): Try[Unit] = synchronized {
+    processConsoleLogs(console)
+
+    val errs = callPop("scalajsPopCapturedErrors").map(_.toString).toList
+
+    cleanupDriver()
+
+    if (errs.nonEmpty) {
+      val msg = ("Errors caught by browser:" :: errs).mkString("\n")
+      Failure(throw new Exception(msg))
+    } else {
+      Success(())
+    }
+  }
+
+  private def cleanupDriver(): Unit = {
```

Comment:
As `stop()`, this used to have `synchronized`, but now it doesn't. Called from `finalize()`, there won't be synchronization. Is that problematic? I don't think so, because I guess `finalize()` can only be called when no one else could call another method anyway. Is that right?